### PR TITLE
cargo-deb: 1.24.0 -> 1.29.1 + build fix

### PR DESCRIPTION
**Pull request by bobrik**
###### Motivation for this change

The `checkType` attribute does not apply after #113193, and since we're changing tests anyway, let's run them in the release mode to avoid the need for `checkType`.

This fixes the otherwise broken build.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [ ] NixOS
   - [x] macOS
   - [x] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [x] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by r-rmcgibbo** [comment]
Result of `nixpkgs-review pr 114311` at 0ae422a4 run on aarch64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package failed to build:</summary>
  <ul>
    <li><a href="https://gist.github.com/0609cf8cefde59ca739da87c6abae8e5">cargo-deb</a></li></li>
  </ul>
</details>

**Comment by bobrik** [comment]
Addressed the Linux build failure.

File: pkgs/tools/package-management/cargo-deb/default.nix [hunk 1]
```diff
@@ -8,24 +8,26 @@
...
   pname = "cargo-deb";
-  version = "1.24.0";
+  version = "1.29.1";
 
   src = fetchFromGitHub {
     owner = "mmstick";
     repo = pname;
-    rev = "b49351f6770aa7aeb053dd1d4a02d6b086caad2a";
-    sha256 = "1hs96yv0awgi7ggpxp7k3n21jpv642sm0529b21hs9ib6kp4vs8s";
+    rev = "86d02f9cacaf4a4f9b576e2dbd9dad65baa61a0d";
```
Comment (SuperSandro2000) [inline pkgs/tools/package-management/cargo-deb/default.nix:16]: Can we use an upstream tag?

File: pkgs/tools/package-management/cargo-deb/default.nix [hunk 2]
```diff
@@ -8,24 +8,26 @@
...
+  cargoSha256 = "0j9frvcmy9hydw73v0ffr0bjvq2ykylnpmiw700z344djpaaa08y";
 
   preCheck = ''
     substituteInPlace tests/command.rs \
-      --replace 'target/debug' "target/${rust.toRustTarget stdenv.buildPlatform}/debug"
+      --replace 'target/debug' "target/${rust.toRustTarget stdenv.buildPlatform}/release"
+
+    # This is an FHS specific assert depending on glibc location
+    substituteInPlace src/dependencies.rs \
+      --replace 'assert!(deps.iter().any(|d| d.starts_with("libc")));' '// no libc assert here'
```
Comment (Mic92) [inline pkgs/tools/package-management/cargo-deb/default.nix:30]: To make this more ryantm-r friendly as the error is only detected at runtime, can we make this a patch instead?

File: pkgs/tools/package-management/cargo-deb/default.nix (hunk 1 above)
Comment (bobrik) [reply pkgs/tools/package-management/cargo-deb/default.nix]: Seems like a good idea, updated.

File: pkgs/tools/package-management/cargo-deb/default.nix (hunk 2 above)
Comment (bobrik) [reply pkgs/tools/package-management/cargo-deb/default.nix:30]: It's detected at build time, during the test. Nothing changes for runtime here.

**Pull request merged**